alacritty/src/renderer/solidrect.rs

```diff
@@ -0,0 +1,200 @@
+use alacritty_terminal::term::SizeInfo;
+
+use crate::gl;
+use crate::gl::types::*;
+use crate::renderer::rects::RenderRect;
+use crate::renderer::RectShaderProgram;
+
+use std::mem::size_of;
+use std::ptr;
+
+const MAX_U16_INDICES: usize = 65536;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Rgba {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Vertex {
+    // TODO these can certainly be i16
+    x: f32,
+    y: f32,
+    color: Rgba,
+}
+
+#[derive(Debug)]
+pub struct SolidRectRenderer {
+    vao: GLuint,
+    vbo: GLuint,
+    ebo: GLuint,
+
+    uploaded_indices: usize,
+    indices: Vec<u16>,
+    vertices: Vec<Vertex>,
+}
+
+impl SolidRectRenderer {
+    pub fn new() -> Self {
+        let mut vao: GLuint = 0;
+        let mut vbo: GLuint = 0;
+        let mut ebo: GLuint = 0;
+
+        unsafe {
+            gl::GenVertexArrays(1, &mut vao);
+            gl::GenBuffers(1, &mut vbo);
+            gl::GenBuffers(1, &mut ebo);
+
+            gl::BindVertexArray(vao);
+            gl::BindBuffer(gl::ELEMENT_ARRAY_BUFFER, ebo);
+
+            gl::BindBuffer(gl::ARRAY_BUFFER, vbo);
+            // Position.
+            gl::VertexAttribPointer(
+                0,
+                2,
+                gl::FLOAT,
+                gl::FALSE,
+                (size_of::<Vertex>()) as _,
+                ptr::null(),
+            );
+            gl::EnableVertexAttribArray(0);
+
+            // Color
+            gl::VertexAttribPointer(
+                1,
+                4,
+                gl::UNSIGNED_BYTE,
+                gl::TRUE,
+                (size_of::<Vertex>()) as _,
+                offset_of!(Vertex, color) as *const _,
+            );
```

Comment:
I think you can look into how we're handing `VertexAttribPointer` without a `offset_of` in `render/mod.rs`, since it seems to work fine for quite a long time already.